### ISSUE TITLE
subcmd function parse normal function

### ISSUE THREAD

**Issue by Zhengqbbb** [jeff-hykin/better-shell-syntax#45]
## The code with a problem is:
```sh
aaa()(
    if [ "$1" = "-h" ] || [ "$1" = "--help" ]; then printf "%s" "help"; return 0;
    fi
)
```

## It looks like: Theme `Vitesse QB`
** screenshot + theme name preferable **
<img width="947" alt="image" src="https://user-images.githubusercontent.com/40693636/218759801-7032f127-52c3-4a17-af63-68befc13c5d9.png">

## It should look like:
<img width="911" alt="image" src="https://user-images.githubusercontent.com/40693636/218759858-36e11d2f-1603-4068-a027-c95346d36dea.png">

**Comment by 1480c1** [comment]
I believe this issue is caused by e07680cb4e7c1f5fa6492a0a94a3643e21f6e9d0, testing the shell.tmLanguage.json from e07680cb4e7c1f5fa6492a0a94a3643e21f6e9d0~ (one commit behind 7c451e4), I get the expected

![image](https://user-images.githubusercontent.com/8345542/223309480-927bebed-b377-43d4-a1bc-2e0a067765e2.png)

but testing e07680cb4e7c1f5fa6492a0a94a3643e21f6e9d0 I get

![image](https://user-images.githubusercontent.com/8345542/223309571-f1757a70-7ba0-431a-8afc-48e96d552058.png)

I can get subshell functions to highlight again after adding
```diff
diff --git a/autogenerated/shell.tmLanguage.json b/autogenerated/shell.tmLanguage.json
index 4cbc795..7a4422f 100644
--- a/autogenerated/shell.tmLanguage.json
+++ b/autogenerated/shell.tmLanguage.json
@@ -555,6 +555,26 @@
               "include": "#initial_context"
             }
           ]
+        },
+        {
+          "begin": "\\(",
+          "end": "\\)",
+          "beginCaptures": {
+            "0": {
+              "name": "punctuation.definition.group.shell punctuation.section.function.definition.shell"
+            }
+          },
+          "endCaptures": {
+            "0": {
+              "name": "punctuation.definition.group.shell punctuation.section.function.definition.shell"
+            }
+          },
+          "name": "meta.function.body.shell",
+          "patterns": [
+            {
+              "include": "#initial_context"
+            }
+          ]
         }
       ]
     },
```
but it seems very hacky? I do not have a good grasp yet on textmate etc. since I only just learned of its existence through this issue

**Comment by 1480c1** [comment]
As a note, my setup is
```none
Version: 1.76.0 (system setup)
Commit: 92da9481c0904c6adfe372c12da3b7748d74bdcb
Date: 2023-03-01T10:22:44.506Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes
```
and I tested it with extensions disabled (so I think no extensions should be affecting .sh files)

**Comment by jeff-hykin** [comment]
> but it seems very hacky? I do not have a good grasp yet on textmate

Textmate is just another word for "the hackiest of hacks", it is truely a terrible parsing engine. For most languages its provably impossible to correctly parse the language using textmate. So its just a matter of limiting the broken cases.

**Comment by jeff-hykin** [comment]
Should now be fixed with v1.4.3!

and just as an example of the hacky-ness. The function pattern usually ends with a `lookBehindFor("}")` but adding a `lookBehindFor(")")` doesn't work because it immediately finds the `)` of the `func_name()`. So the hack to get around that is to have another pattern match the space/tab/newline thats after the `func_name()` which blocks the `lookBehindFor(")")` for one char before allowing it to run.

**Comment by 1480c1** [comment]
Thanks! I can confirm that this is fixed with the latest version, and you've indeed convinced me that it's all just a giant hack, haha. Language parsing does not seem like the easiest thing to get right the first time, nor in a way that will never seem hacky.

![image](https://user-images.githubusercontent.com/8345542/223506774-f7b4102a-eace-4498-be8e-e611b8ad1e9c.png)
